src/LineageTree/lineageTree.py

```diff
@@ -2056,7 +2060,14 @@ def unordered_tree_edit_distances_at_time_t(
         elif t in self.uted and not recompute:
             return self.uted[t]
         self.uted[t] = {}
-        roots = self.time_nodes[t]
+        if specific_roots:
+            roots = [
+                i
+                for i in self.time_nodes[t]
+                if self.get_ancestor_at_t(i) in set(specific_roots)
+            ]
+        else:
+            roots = self.time_nodes[t]
```

Comment:
Here wouldn't it be faster to take all the specific roots and go down from them to time point `t` and get all the cells?

Something like that:
```python
roots = []
for r in specific_roots:
    roots += self.get_cells_at_t_from_root(r, t)
```
That way your specific roots don't even have to be at the first time point neither all at the same time points.

I don't think `get_cells_at_t_from_root` is implemented but it could be something like that:
```python
def get_cells_at_t_from_root(r: int, t: int = None) -> list:
    """Returns the list of cells at time `t` that are spawn by the node `r`.

   Args:
      r (int): id of the spawning node
      t (int): target time, if None goes as far as possible
            (default None)

   Returns:
      (list) list of nodes at time `t` spawned by `r`
    """
   to_do = [r]
   final_nodes = []
   while 0<len(to_do):
      curr = to_do.pop()
      for next in self[curr]:
         if self.time[next] < t:
            to_do.append(next)
         elif self.time[next] == t:
            final_nodes.append(next)
    return final_nodes
```
Not sure it works, haven't tested it but should be something along these lines :D

Reply:
I will check which one works faster and I will keep the fastest way!

Reply:
```python
def get_cells_at_t_from_root(self, r: int, t: int = None) -> list:
    """Returns the list of cells at time `t` that are spawn by the node `r`.

   Args:
      r (int): id of the spawning node
      t (int): target time, if None goes as far as possible
            (default None)

   Returns:
      (list) list of nodes at time `t` spawned by `r`
    """
   to_do = [r]
   final_nodes = []
   while 0<len(to_do):
      curr = to_do.pop()
      for next in self[curr]:
         if self.time[next] < t:
            to_do.append(next)
         elif self.time[next] == t:
            final_nodes.append(next)
    return final_nodes
```

Reply:
I still get the same error  'NoneType' object is not iterable